pkg/config/store/queue_test.go

```diff
@@ -139,19 +140,27 @@ func TestQueueCancelSync(t *testing.T) {
 			Value: &BackEndResource{},
 		}
 	}
+	// Wait for the queue's run loop to process chin.
+	time.Sleep(time.Millisecond)
 	cancel()
 	// Wait for the queue's run loop to end.
 	time.Sleep(time.Millisecond)
 	// Read the bufferred events.
+	emptyEv := Event{}
 	for i := 0; i < choutBufSize; i++ {
 		ev := <-q.chout
+		if reflect.DeepEqual(ev, emptyEv) {
+			// ev can be empty; this happens when run() goroutine doesn't run enough
+			// before cancel() is called.
+			break
```

Comment:
If we break here, does it mean we didn't test the thing at all ?

Reply:
It should break here because `q.chout` is considered to be closed already. This is the core part to address the error you've met. Here's the thing; `q.chout` is a channel with some size of buffer, and `q` maintains a goroutine to push events into `chout`. This test pushes sufficient events, and then cancel the context, and check the content in the `chout`.
The existing code assumes `chout`'s buffer is filled with the created events, but sometimes goroutine can be stopped before putting the events fully. In that case it should break the loop.